src/core_plugins/kibana/public/visualize/editor/styles/_editor.less

```diff
@@ -287,6 +287,17 @@
       padding: @vis-editor-agg-editor-spacing;
     }
 
+  .vis-editor-field-ui-select .ui-select-match .btn {
```

Comment:
You're right, we do want to avoid creating long hyper-specific selectors where possible, but in this case there's no way around it since we don't control the markup. So I think the best we can do is try to make the relationships of the classes as clear as possible.